lab_2_keywords_cooccurrence/main.py

```diff
@@ -120,10 +189,23 @@ def calculate_cumulative_score_for_candidates(candidate_keyword_phrases: KeyPhra
 
     In case of corrupt input arguments, None is returned
     """
-    if not type_check(candidate_keyword_phrases, list) or not type_check(word_scores, dict) or \
-            not all(token in word_scores for token in list(chain.from_iterable(candidate_keyword_phrases))):
+
+    if not isinstance(candidate_keyword_phrases, list) or not candidate_keyword_phrases:
+        return None
+    if not isinstance(word_scores, dict) or not word_scores:
+        return None
+    if len(candidate_keyword_phrases) == 0 or len(word_scores) == 0:
         return None
-    return {phrase: sum(word_scores[token] for token in phrase) for phrase in candidate_keyword_phrases}
+    for phrase in candidate_keyword_phrases:
+        for word in phrase:
+            if word not in word_scores:
+                return None
+    cumulative_score_for_candidates = {}
+    for phrase in candidate_keyword_phrases:
+        cumulative_score_for_candidates[phrase] = 0.0
```

Comment:
what if `phrase` is already in `cumulative_score_for_candidates` with not 0 value?